06_puppy/pfannerj/storer_test.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"testing"
+
+	tassert "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storerImpl int
+
+const (
+	syncImpl storerImpl = iota
+	mapImpl
+)
+
+const firstPuppyID uint32 = 1
+
+var (
+	firstPuppy = func() Puppy {
+		return Puppy{
+			Breed:  "Retriever",
+			Colour: "Golden",
+			Value:  9999.99,
+		}
+	}
+	modifiedPuppy = func() Puppy {
+		return Puppy{
+			Breed:  "Bitsa",
+			Colour: "Mixed",
+			Value:  1.99,
+		}
+	}
+	anotherPuppy = func() Puppy {
+		return Puppy{
+			Breed:  "Labrador",
+			Colour: "Black",
+			Value:  999.99,
+		}
+	}
+)
+
+type storerSuite struct {
+	suite.Suite
+	store Storer
+	impl  storerImpl
+}
+
+func (s *storerSuite) SetupTest() {
+	var err error
+	switch s.impl {
+	case syncImpl:
+		// create a new syncstore and add the first puppy
+		s.store = NewSyncStore()
+		puppySyncImpl := firstPuppy()
+		_, err = s.store.CreatePuppy(&puppySyncImpl)
+	case mapImpl:
+		// create a new mapstore and add the first puppy
+		s.store = NewMapStore()
+		puppyMapImpl := firstPuppy()
+		_, err = s.store.CreatePuppy(&puppyMapImpl)
+	default:
+		panic("Unrecognised storer implementation")
+	}
+	if err != nil {
+		panic("Failed to setup puppy test")
+	}
+}
+
+func (s *storerSuite) TestCreate() {
+	// given
+	assert := tassert.New(s.T())
+	newPuppy := anotherPuppy()
+
+	// when
+	createdPuppyID, err := s.store.CreatePuppy(&newPuppy)
+	newPuppy.ID = createdPuppyID
+
+	// then
+	assert.NoError(err, "Should not get an error creating a puppy")
+	foundPuppy, err := s.store.ReadPuppy(createdPuppyID)
+	if assert.NoError(err, "Should be able to read an newly created puppy") {
+		assert.Equal(&newPuppy, foundPuppy, "Created puppy should be identical to the one passed to create")
+	}
```

Comment:
```go
assert.NoError(err, "Should be able to read an newly created puppy")
assert.Equal(&newPuppy, foundPuppy, "Created puppy should be identical to the one passed to create")
```

same below. 
I don't understand the`if assert.NoError` - do you expect an error or not?

Reply:
Expecting the read to be successful and the newPuppy to equal the foundPuppy... Although I guess maybe the 'if' is not required?....